src/full/Agda/Utils/Pretty/ANSI.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE CPP #-}
+module Agda.Utils.Pretty.ANSI where
+
+import Control.Monad.IO.Class
+import Control.Monad
+
+import Text.PrettyPrint.Annotated.HughesPJ (renderDecorated)
+
+import Agda.Interaction.Options.HasOptions
+import Agda.Interaction.Options.Base
+import Agda.Utils.Pretty.Aspect
+import Agda.Utils.Pretty
+import Agda.Utils.Monad
+
+#ifdef HAS_ISATTY
+import System.Posix.Terminal
+import System.Posix.IO
+#endif
```

Comment:
Can we avoid `#ifdef HAS_ISATTY` (and a complication of the `Agda.cabal` file) here?  There is e.g.
https://hackage.haskell.org/package/pretty-terminal-0.1.0.0/docs/System-Console-Pretty.html#v:supportsPretty

Reply:
If it's alright for me to add a dependency on a non-boot library, then sure, but I'd like to use this instead: https://hackage.haskell.org/package/ansi-terminal-0.11.4/docs/System-Console-ANSI.html

It's maintained, and it also supports Windows, which would mean I can implement automatic colouring there, too. There's a minor downside (having to "render" directly in IO) but it would be totally localized.